### ISSUE TITLE
Cluster icon not disappearing on mobile browsers

### ISSUE THREAD

**Issue by fabriziomazzoni** [Leaflet/Leaflet.markercluster#655]
When clicking on a cluster and zooming in, the round cluster icon will not disappear. This behaviour happens only on mobile browsers and not desktop. Tested on android browser, chrome android and opera android

**Comment by danzel** [comment]
Is this a dupe of #529?

**Comment by EsaKarvanen** [comment]
I don't know if this is the same problem I'm having but zooming in mobile browsers (at least Chrome for Android and Safari in iOS) often leaves clusters visible even they should disappear. I debugged a little and saw this:

_moveEnd starts
MarkerCluster _recursivelyRemoveChildrenFromMap zoomLevel: 11
MarkerCluster _recursivelyAddChildrenToMap zoomLevel: 10
_moveEnd ends
_zoomEnd starts
_mergeSplitClusters
_mergeSplitClusters Zoom out, merge
_noAnimation _animationZoomOut previousZoomLevel: 11, newZoomLevel: 9
MarkerCluster _recursivelyRemoveChildrenFromMap zoomLevel: 11
MarkerCluster _recursivelyAddChildrenToMap zoomLevel: 9
_mergeSplitClusters ends
_zoomEnd ends

So, I think that first the zooming starts from level 11 but when in level 10, we receive a move event. In the move event we remove clusters from level 11 and add them back to level 10. Afterwards the zoom event ends in level 9. Now we try to remove clusters from level 11 but they are not there as they were added to level 10.

I was able to get rid of this problem by changing line
this._topClusterLevel._recursivelyAddChildrenToMap(null, Math.round(this._map._zoom), newBounds);

from _moveEnd()
to:
this._topClusterLevel._recursivelyAddChildrenToMap(null, Math.round(this._zoom), newBounds);

Of course I'm not sure what kind of problems that might cause.
